packages/rum-core/src/performance-monitoring/transaction.js

```diff
@@ -138,21 +143,16 @@ class Transaction extends SpanBase {
   }
 
   addTask(taskId) {
-    if (typeof taskId === 'undefined') {
-      taskId = 'task' + this.nextAutoTaskId++
-    }
-    if (this._scheduledTasks.indexOf(taskId) == -1) {
-      this._scheduledTasks.push(taskId)
-      return taskId
+    if (!taskId) {
+      taskId = 'task-' + generateRandomId(16)
```

Comment:
I think we should just use incremental ids here. using random Ids seems like an overkill.

Reply:
This happens only in corner case when user doesn't provide any valid id. Its not overkill IMO, better than not having to store instance level variables.